recipe.py

```diff
@@ -34,9 +34,13 @@ def get_ingredients(self):
     
     def set_ingredients(self,ingredients):
         self._ingredients.append(ingredients)
-
+        if len(set(self._ingredients)) != len(self._ingredients):
+            self._ingredients.remove(ingredients) #I have tested this and it only removes the recently added ingredient
```

Comment:
If its a straightforward set then it should be a simple assignment I think - will probably update when the ingredients class is added in.

Reply:
I am getting confused between if you mean set the data type and set the function? Due to ingredients class does this mean only the self._ingredients.append(ingredients) is needed for this function?

Reply:
So the function is "set_ingredients" a typical interpretation could be that this sets either a subset or all of the ingredients. If you set a subset then this is either modifying a subset of ingredients that are already present or adding a subset (with an expectation of adding more) - or in the other case you are setting all of them. If you want to modify a subset of items in a list say what is typically done is you get the current list (as a copy) modify the items you need to (keeping everything else the same) and then overwrite the original list with the modified one you create. If the list is long and complicated then yes this can become cumbersome and a more sophisticated method of *in place" replacement is required. In this case I don't think the case really requires it. If everything needs to be set then one way or another the original list will be replaced completely so piecemeal partial modification doesn't make sense - it's literally x = new_value (and scrap anything that was in x).

Reply:
ok, i will change set ingredients to set x=newvalue and add an add ingredient function which appends